lib/src/impl_js/impl_js.random.dart

```diff
@@ -17,7 +17,7 @@ part of impl_js;
 void fillRandomBytes(TypedData destination) {
   try {
     subtle.getRandomValues(destination);
-  } on subtle.DomException catch (e) {
+  } on subtle.JSDomException catch (e) {
```

Comment:
Does this work?

Reply:
Can we make some unit tests for `impl_js`.

Like:
```js
crypto.getRandomValues(new Uint8Array(1000000))
// throws:
// Uncaught DOMException: Failed to execute 'getRandomValues' on 'Crypto': The ArrayBufferView's byte length (1000000) exceeds the number of bytes of entropy available via this API (65536).
```

at-least in chrome.


-----------------


Are there scenarios where we could get `TypeError` or other such things when calling into JS, and would handle them correctly?

Reply:
I added a test file for each browser since the error is expressed differently in chrome and firefox. And, I was also able to add a scenario where DOMException is try-catch (a bit weird).
Does this look good?

* [chrome](https://github.com/google/webcrypto.dart/pull/86/files#diff-627e300589c395ca34cb9c192df9b8ea491ed4b16f67cdeaa9b90cd50a4cd441)
* [firefox](https://github.com/google/webcrypto.dart/pull/86/files#diff-8f2af7a8565ec85972baa2d7444e5765368cc42f8905f46df6cac4c6ab2b942d)

ref https://pub.dev/packages/test#restricting-tests-to-certain-platforms

Reply:
Hmm, I guess this works :rocket:

I'm just getting used to that everything returned or thrown by these APIs is in fact a `JSObject`. And any instance of `JSObject` is also an instance of `subtle.JSDomException`.

So any JS exception thrown will be a `subtle.JSDomException`. But that's probably fine, since we always read the `.name` property in order to decide what to do.